libs/components/autonumeric/src/lib/modules/autonumeric/autonumeric.directive.ts

```diff
@@ -165,53 +177,57 @@ export class SkyAutonumericDirective
   }
 
   public registerOnChange(fn: (value: number | undefined) => void): void {
-    this.onChange = fn;
+    this.#onChange = fn;
   }
 
   public registerOnTouched(fn: () => void): void {
-    this.onTouched = fn;
+    this.#onTouched = fn;
   }
 
   @HostListener('blur')
   public onBlur(): void {
-    this.onTouched();
+    this.#onTouched();
   }
 
-  private getNumericValue(): number | undefined {
-    const inputValue = this.getInputValue();
-    return inputValue && !this.isInputValueTheCurrencySymbol(inputValue)
-      ? this.autonumericInstance.getNumber()
+  #getNumericValue(): number | undefined {
+    const inputValue = this.#getInputValue();
+    const result = this.#autonumericInstance.getNumber();
+    const numericValueOrUndefined = !result ? undefined : result;
```

Comment:
I would just add `&& !!result` to the opening portion of the ternary - then you won't need the extra ternary that you added and can just return back `? result : undefined`

Reply:
good suggestion. fixed in: [simplify function](https://github.com/blackbaud/skyux/pull/606/commits/29379133a5c7980e92e027509639cb41e69accca)